pytorch_lightning/trainer/trainer.py

```diff
@@ -449,17 +449,52 @@ def default_attributes(cls):
         return args
 
     @classmethod
-    def add_argparse_args(cls, parent_parser: ArgumentParser) -> ArgumentParser:
-        """Extend existing argparse by default `Trainer` attributes."""
-        parser = ArgumentParser(parents=[parent_parser], add_help=False)
+    def _get_argparse_args_and_types(cls) -> Generator[Tuple[str, Any, Any], None, None]:
```

Comment:
is there an advantage to have it as a generator instead of list/tuple?

Reply:
Yes, you are right. Fixed